extensions/hibernate-search-elasticsearch/runtime/src/main/java/io/quarkus/hibernate/search/elasticsearch/runtime/HibernateSearchElasticsearchRecorder.java

```diff
@@ -124,7 +124,7 @@ private void contributeBackendBuildTimeProperties(BiConsumer<String, Object> pro
         private void contributeBackendRuntimeProperties(BiConsumer<String, Object> propertyCollector, String backendName,
                 ElasticsearchBackendRuntimeConfig elasticsearchBackendConfig) {
             addBackendConfig(propertyCollector, backendName, ElasticsearchBackendSettings.HOSTS,
-                    elasticsearchBackendConfig.hosts,
+                    elasticsearchBackendConfig.hosts.orElse(null),
                     v -> (!v.isEmpty() && !(v.size() == 1 && v.get(0).isEmpty())), Function.identity());
```

Comment:
Looks like there's a hack here that I didn't see last time... My suggestion was wrong, it would trigger an NPE.

I would need to know what happens when the user sets the property to:

1. no value at all (the property is not set)
2. an empty string
3. a blank string (spaces only)

If the answer is "you get an `Optional.empty()` in all three cases", then we can replace this code with:

```
             addBackendConfig(propertyCollector, backendName, ElasticsearchBackendSettings.HOSTS,
                    elasticsearchBackendConfig.hosts); 
```

If the answer is "you get `Optional.empty()` or an `Optional.of(Collections.emptyList())`, depending on the case", then we need the code you submitted in your original PR (sorry...):

```
             addBackendConfig(propertyCollector, backendName, ElasticsearchBackendSettings.HOSTS,
                    elasticsearchBackendConfig.hosts.orElse(Collections.emptyList()),
                    v -> (!v.isEmpty() && !(v.size() == 1 && v.get(0).isEmpty())), Function.identity()); 
```

If the answer is "you get `Optional.empty()` or `Optional.of(Collections.emptyList())`, or `Optional.of(Arrays.asList("    ")))`, respectively", then we should add some post-processing, but the code just above would be good enough for now.

Reply:
You will always get `Optional.empty()` in all three cases.  I've made the suggested change.